### PR TITLE
Create new version 2022-09-30-preview for MySQL

### PR DESCRIPTION
**Pull request by xiaoxuqi-ms**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raosuhas** [comment]
> Hi, @xiaoxuqi-ms, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)

@xiaoxuqi-ms can you please follow the instructions here when creating a new api version ?

**Comment by raosuhas** [comment]
Can you also please fix the breaking change s pointed out in the PR ?

**Comment by xiaoxuqi-ms** [comment]
> > Hi, @xiaoxuqi-ms, For review efficiency consideration, when creating a new api version, it is required to place API specs of the base version in the first commit, and push new version updates into successive commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Or you could onboard [API spec pipeline](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/606/Enable-API-Spec-Pipeline)
> 
> @xiaoxuqi-ms can you please follow the instructions here when creating a new api version ?

Hi @raosuhas , thanks for reviewing my pr, and this pr follows the instruction for new api version: 
image
Could you help to review again?

**Comment by xiaoxuqi-ms** [comment]
The linter errors are for the existing apis, and to fix the linter issues may cause breaking changes since the response type of API call is changed. So we prefer not to fix the linter errors for the existing APIs.

**Comment by weidongxu-microsoft** [comment]
I may ignore the LRO related LintDiff (most SDK people don't like `x-ms-long-running-operation-options` at all). But you still need to fix others, and pass ARM.

**Comment by xiaoxuqi-ms** [comment]
Hi @weidongxu-microsoft , thanks a lot for reviewing this pr. For linter issues, except the LRO option, there is still one error: add Location in Header. I added the Location but the response changed to 'Microsoft.Rest.Azure.AzureOperationResponse<Microsoft.RDMS.ResourceManagement.Client.Models.Server, Microsoft.RDMS.ResourceManagement.Client.V20220930.Models.ServersUpdateHeaders>' from Microsoft.Rest.Azure.AzureOperationResponse<Microsoft.RDMS.ResourceManagement.Client.Models.Server>'. I am afraid this would be a breaking change to customers, Since we did not update the currenting API, can we ignore the error?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/Backups/preview/2022-09-30-preview/BackupAndExport.json [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+        "operationId": "BackupAndExport_ValidateBackup",
+        "x-ms-examples": {
+          "Validate Backup": {
+            "$ref": "./examples/ValidateBackup.json"
+          }
+        },
+        "description": "Validates if backup can be performed for given server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
```
Comment (rkmanda) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/Backups/preview/2022-09-30-preview/BackupAndExport.json:106]: >v2 [](http://example.com/codeflow?start=72&length=2)

use the latest relevant version

Comment (xiaoxuqi-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/Backups/preview/2022-09-30-preview/BackupAndExport.json:106]: ApiVersionParameter and ResourceGroupNameParameter will use v5, but other definition will use v2, in our e2e test, we still use a lower version autorest to generate SDK which does not support "format"

**Comment by xiaoxuqi-ms** [comment]
Hi @rkmanda , thank you for reviewing my pr. For linter issues: 
1. final-state-via: in header, we use location to trace final state, so I set final-state-via to location
2. For status codes, since our backend service has frozen for this release, so I cannot remove/update any status codes. I will discuss them with feature owners and maybe they can fixed in next version. May I have an exception for these errors in this pr?

**Comment by xiaoxuqi-ms** [comment]
There are several commits pushed after breaking change approval and I believe it is necessary to request a breaking change review again.
The new changes are as following:
1.	For service operation, we will support LocationBasedCapabilities_Get, which can be used as a new API, the operationId is different.
2.	Default value changes: to fix roundtrip validation
3.	Readonly property changes: to fix roundtrip validation
4.	The removed properties in common-types are false alarm.
5.	The removed properties in service operation are false alarm.
6.	The x-ms-enum name change is to fix roundtrip validation, I need more advice for this. [@Bin Xu](mailto:bixu@microsoft.com)

**Review by ihsanali01 — APPROVED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/Backups/preview/2022-09-30-preview/BackupAndExport.json [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+            "Pending",
+            "InProgress",
+            "Succeeded",
+            "Failed",
+            "CancelInProgress",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "OperationStatus",
+            "modelAsString": false,
```
Comment (zizw123) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/Backups/preview/2022-09-30-preview/BackupAndExport.json:274]: >false [](http://example.com/codeflow?start=29&length=5)

Why marking as false? May potential break client if changes are introduced in future

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json [hunk 3]
```diff
@@ -0,0 +1,1232 @@
...
+              "$ref": "#/definitions/Server"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/Server"
+            }
+          },
+          "202": {
```
Comment (zizw123) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json:94]: >202 [](http://example.com/codeflow?start=11&length=3)

202 is old model, see [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
```
Comment (zizw123) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json:610]: This is an optional but good to have, just curious to see why this is only added here but not for the above POST

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json [hunk 5]
```diff
@@ -0,0 +1,1232 @@
...
+        "type": {
+          "type": "string",
+          "description": "The key type, AzureKeyVault for enable cmk, SystemManaged for disable cmk.",
+          "enum": [
+            "AzureKeyVault",
+            "SystemManaged"
+          ],
+          "x-ms-enum": {
+            "name": "DataEncryptionType",
+            "modelAsString": false
```
Comment (zizw123) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json:726]: >false [](http://example.com/codeflow?start=29&length=5)

Same question for this, can this be true?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json (hunk 4 above)
Comment (xiaoxuqi-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/FlexibleServers/preview/2022-09-30-preview/FlexibleServers.json:610]: Beacuse this is a new api, and we do not update the existing api to aovoid breaking change or unnecessary changes

**Comment by zizw123** [comment]
Signing off, please make sure to open a track item for next version to fix the 202

File: custom-words.txt [hunk 6]
```diff
@@ -2785,4 +2785,8 @@ Unprepares
 Unpreparing
 typespec
 DSPM
-Dspm
\ No newline at end of file
+Dspm
+Gtid
+GTID
+gtid
+Dspm
```
Comment (weidongxu-microsoft) [inline custom-words.txt:2792]: two Dspm?

And where are them used?

Comment (xiaoxuqi-ms) [reply custom-words.txt:2792]: this is caused my resolving conflicts. Let me fix

**Comment by weidongxu-microsoft** [comment]
I cannot find a proper diff on 2022-09-30-preview/FlexibleServers.json, could you provide one for easy to review what is changed compared to previous preview?

Also  you need to fix Avocado.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**